### PR TITLE
Update SkyDNS and disable recursion

### PR DESCRIPTION
**Pull request by sosiouxme**
Address https://github.com/openshift/origin/issues/2482#issuecomment-106372072

### CODE REVIEW

**Comment by sosiouxme** [comment]
@ncdc PTAL?

File: pkg/cmd/server/origin/master.go [hunk 1]
```diff
@@ -776,6 +776,7 @@ func (c *MasterConfig) RunDNSServer() {
 		glog.Fatalf("Could not start DNS: %v", err)
 	}
 	config.DnsAddr = c.Options.DNSConfig.BindAddress
+	config.NoRec = true // do not want to deploy an open resolver
```
Comment (ncdc) [inline pkg/cmd/server/origin/master.go]: I put this in NewServerDefaults in pkg/dns/server.go... seemed to make more sense to me there, but I'm not sure what the distinction between the 2 is supposed to be.

Comment (sosiouxme) [inline pkg/cmd/server/origin/master.go]: This looked like a location where in the future we could read more options from c.Options.DNSConfig and set them on the skydns config.

**Comment by sosiouxme** [comment]
[test]

**Comment by sosiouxme** [comment]
EC2 fail, [test] again

**Comment by ncdc** [comment]
[test]

**Comment by sosiouxme** [comment]
[test] once more

**Comment by sosiouxme** [comment]
@smarterclayton any objection or ready for merge?

**Comment by smarterclayton** [comment]
This is fine with me if Andy signs off

**Comment by smarterclayton** [comment]
Since Andy's out and no comments, LGTM [merge]

**Pull request merged**